fili-core/src/main/java/com/yahoo/bard/webservice/table/ConcretePhysicalTable.java

```diff
@@ -37,26 +35,29 @@ public ConcretePhysicalTable(
                 timeGrain,
                 columns,
                 logicalToPhysicalColumnNames,
-                new ImmutableAvailability(factTableName, Collections.emptyMap())
+                new ImmutableAvailability(name, Collections.emptyMap())
         );
     }
 
     /**
      * Create a concrete physical table.
      * The fact table name will be defaulted to the name and the availability initialized to empty intervals.
      *
-     * @param name  Fili name of the physical table
+     * @param name  Name of the physical table as String
      * @param timeGrain  time grain of the table
      * @param columns The columns for this table
      * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     *
+     * @deprecated Should use constructor with TableName instead of String as table name
      */
+    @Deprecated
     public ConcretePhysicalTable(
             @NotNull String name,
             @NotNull ZonedTimeGrain timeGrain,
             @NotNull Iterable<Column> columns,
             @NotNull Map<String, String> logicalToPhysicalColumnNames
     ) {
-        this(name, name, timeGrain, columns, logicalToPhysicalColumnNames);
+        this(TableName.of(name), timeGrain, columns, logicalToPhysicalColumnNames);
```

Comment:
getFactTableName should be serving from a local copy of the name provided in construction, not pried out of the availability.